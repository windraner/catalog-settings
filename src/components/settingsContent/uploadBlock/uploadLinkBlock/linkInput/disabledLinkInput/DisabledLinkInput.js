import React, { Component } from 'react';
import ToolTip from '../../../toolTip/ToolTip';

import './disabledLinkInput.css';

export default class DisabledLinkInput extends Component {
  state = {
    isTooltipShow: false
  }

  hoverHandler = (value) => {
    this.setState({isTooltipShow: value});
  }

  render() {
    const { isTooltipShow } = this.state;

    return (
      <div
        onMouseEnter={() => this.hoverHandler(true)}
        onMouseLeave={() => this.hoverHandler(false)}
        className="catalog-settings-content__link-input catalog-settings-content__link-input_disabled"
      >
        <div className="catalog-settings-content__tooltip-wrapper">
          <ToolTip
            isShow={isTooltipShow}
            text={'Remove uploaded file to add link to CSV source file'}
          />
        </div>
        http://mysite.com/patch/to/CSV/source.csv
      </div>
    );
  }
}
