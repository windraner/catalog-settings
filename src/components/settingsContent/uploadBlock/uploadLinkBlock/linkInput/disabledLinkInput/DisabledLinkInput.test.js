import React from 'react';
import DisabledLinkInput from './DisabledLinkInput';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('<DisabledLinkInput />', () => {
  const wrapper = shallow(<DisabledLinkInput />)
  it('Check for className and text', () => {
    expect(wrapper.find('.catalog-settings-content__link-input').length).toBe(1);
    expect(wrapper.find('.catalog-settings-content__link-input').text()).toBe('<ToolTip />http://mysite.com/patch/to/CSV/source.csv');
  });
  it('Check hover change state', () => {
    wrapper.find('.catalog-settings-content__link-input').simulate("mouseEnter");
    expect(wrapper.state('isTooltipShow')).toBe(true);
    wrapper.find('.catalog-settings-content__link-input').simulate("mouseLeave");
    expect(wrapper.state('isTooltipShow')).toBe(false);
  });
  it('matches the snapshot', () => {
    const tree = shallow(<DisabledLinkInput />)
    expect(toJson(tree)).toMatchSnapshot()
  })
});
