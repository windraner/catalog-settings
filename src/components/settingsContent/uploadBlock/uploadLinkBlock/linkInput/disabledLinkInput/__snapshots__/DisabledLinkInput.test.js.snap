// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisabledLinkInput /> matches the snapshot 1`] = `
<div
  className="catalog-settings-content__link-input catalog-settings-content__link-input_disabled"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="catalog-settings-content__tooltip-wrapper"
  >
    <ToolTip
      isShow={false}
      text="Remove uploaded file to add link to CSV source file"
    />
  </div>
  http://mysite.com/patch/to/CSV/source.csv
</div>
`;
